File: leetcode/subArraySum.py
class Solution:
    import bisect
    def subarraySum(self, nums: List[int], k: int) -> int:
        vis = {}
        vis[0] = 1
        cnt = 0
        now = 0
        for i in range(len(nums)):
            now += nums[i]
            if now - k in vis:
                cnt += vis[now-k]
            if now not in vis:vis[now] = 0
            vis[now] += 1
        return cnt

File: leetcode/subArraySum.go
package main

func subarraySum(nums []int, k int) int {
	now := 0
	res := 0
	numCounts := make(map[int]int)
	numCounts[0] = 1
	for _, v := range nums {
		now += v
		res += numCounts[v-k]
		numCounts[v] += 1
	}
	return res
}
